Source/WebCore/loader/cache/CachedFontLoadRequest.h

```diff
@@ -81,14 +81,14 @@ class CachedFontLoadRequest final : public FontLoadRequest, public CachedFontCli
 
     void fontLoaded(CachedFont& font) final
     {
-        ASSERT_UNUSED(font, &font == m_font.get());
-        if (m_fontLoadRequestClient)
-            m_fontLoadRequestClient->fontLoaded(*this);
-
         if (m_font->didRefuseToLoadCustomFont() && m_context) {
             auto message = makeString("[Lockdown Mode] This font has been blocked: ", m_font->url().string());
             m_context->addConsoleMessage(MessageSource::Security, MessageLevel::Info, message);
         }
+
+        ASSERT_UNUSED(font, &font == m_font.get());
+        if (m_fontLoadRequestClient)
+            m_fontLoadRequestClient->fontLoaded(*this); // fontLoaded() might destroy This object. Don't deref its members after it.
     }
 
     CachedResourceHandle<CachedFont> m_font;
```

Comment:
CachedResourceHandle should probably hold a WeakPtr to the font.

Reply:
I don't understand this comment. A CachedResourceHandle is essentially a RefPtr but for CachedResources objects.

Reply:
Maybe we should have a WeakPtr (Or CheckedRef) to CSSFontFace instead of a l-value reference, so CSSFontFaceSource can check that its FontFace owner is already destroyed for some reason and abort the loading process?

Reply:
We're commenting on a line with `CachedResourceHandle<CachedFont>`. And you're talking about CSSFontFaceSource & a FontFace objects. I have trouble following.

Reply:
`CachedFontLoadRequest::fontLoaded` calls `CSSFontFaceSource::fontLoaded`
```
void CSSFontFaceSource::fontLoaded(FontLoadRequest& fontRequest)
{
    ...
    Ref<CSSFontFace> protectedFace(m_face);
    ...
}
```

We end up calling the destructor of  CSSFontFace once we leave the function’s scope. This means that this Ref was the only one to the FontFace, but there was no way for us to know that, since we have a l-value ref (m_face ) to it here.